src/librustc/middle/cstore.rs

```diff
@@ -474,7 +477,13 @@ pub mod tls {
     {
         let tls_payload = (ecx as *const _, encoder as *mut _);
         let tls_ptr = &tls_payload as *const _ as *const TlsPayload;
-        TLS_ENCODING.set(unsafe { &*tls_ptr }, || f(ecx, encoder))
+        TLS_ENCODING.with(|tls| {
+            let prev = tls.get();
```

Comment:
On Fri, Feb 26, 2016 at 04:16:52PM -0800, Alex Crichton wrote:
> > @@ -474,7 +477,13 @@ pub mod tls {
> >      {
> >          let tls_payload = (ecx as *const _, encoder as *mut _);
> >          let tls_ptr = &tls_payload as *const _ as *const TlsPayload;
> > -        TLS_ENCODING.set(unsafe { &*tls_ptr }, || f(ecx, encoder))
> > +        TLS_ENCODING.with(|tls| {
> > +            let prev = tls.get();
> 
> It's repeated precisely three times :) (the three times scoped tls was used)
> 
> I wouldn't mind adding a rustc-specific abstraction for this though if you'd like

I had assumed we would make a rustc-specific abstraction, yes, but I think the
current stuff is prob fine too